src/shared/udev-util.c

```diff
@@ -192,7 +210,7 @@ int device_wait_for_initialization(sd_device *device, const char *subsystem, use
 
         /* Check again, maybe things changed. Udev will re-read the db if the device wasn't initialized
          * yet. */
-        if (sd_device_get_is_initialized(device) > 0) {
+        if (device && sd_device_get_is_initialized(device) > 0) {
```

Comment:
we need this check for the case where we only get devlink passed in here, too.

i.e. it might be that while we are setting up the monitor and event loop the device appears. and hence we enter the event loop thinking the device is still missing, even though it just appeared.

i.e. we should try to create an sd_device object here from the devlink, and if it exists then check sd_device_is_initialized(), and if so return that. i.e. basically, if the device appeared here, then enter the traditional code path where we watch for an existing device to finish initializing.

Reply:
Uhm maybe I'm missing something, but I don't see a constructor that can use just the devlink for sd_device?

```
int sd_device_new_from_syspath(sd_device **ret, const char *syspath);
int sd_device_new_from_devnum(sd_device **ret, char type, dev_t devnum);
int sd_device_new_from_subsystem_sysname(sd_device **ret, const char *subsystem, const char *sysname);
int sd_device_new_from_device_id(sd_device **ret, const char *id);
```

I'll do this with the access() check for now

Reply:
```c
stat(p, &st) + sd_device_new_from_devnum(&d, 'b', st.st_rdev);
```

Reply:
thanks, done